joft/utils.py

```diff
@@ -28,8 +28,47 @@ def read_and_validate_config(
     return config
 
 
-def load_toml_app_config() -> typing.Any:
-    possible_paths = []
+def load_toml_app_config(config_path: str | None = None) -> typing.Any:
+    """Loads the TOML application configuration.
+
+    If a config_path is provided, it will be loaded first. Otherwise, it will search for the config file in the default locations.
+
+    Args:
+        config_path: Optional path to the config file.
+
+    Returns:
+        The application configuration as a dictionary.
+
+    Raises:
+        SystemExit: If the configuration file is invalid or not found.
+    """
+    possible_paths: list[str] = []
+
+    if config_path:
+        config_file_path = pathlib.Path(config_path)
+        if not config_file_path.is_file():
+            print(f"[ERROR] Config file not found: {config_path}")
+            sys.exit(1)
+        try:
```

Comment:
This whole part is basically a copy of the code we already have below. Would it make sense to DRY?